src/test/javascript/spec/component/localvc/vcs-repository-access-log-view.component.spec.ts

```diff
@@ -0,0 +1,86 @@
+    let fixture: ComponentFixture<VcsRepositoryAccessLogViewComponent>;
+    let programmingExerciseParticipationService: ProgrammingExerciseParticipationService;
+    const userId = 4;
+    let participationVcsAccessLogSpy: jest.SpyInstance;
+    let repositoryVcsAccessLogSpy: jest.SpyInstance;
+
+    const mockVcsAccessLog: VcsAccessLogDTO[] = [
+        {
+            id: 1,
+            userId: userId,
+            name: 'authorName',
+            email: 'authorEmail',
+            commitHash: 'abcde',
+            authenticationMechanism: 'SSH',
+            repositoryActionType: 'WRITE',
+            timestamp: dayjs('2021-01-02'),
+        },
+        {
+            id: 2,
+            userId: userId,
+            name: 'authorName',
+            email: 'authorEmail',
+            commitHash: 'fffee',
+            authenticationMechanism: 'SSH',
+            repositoryActionType: 'READ',
+            timestamp: dayjs('2021-01-03'),
+        },
+    ];
+
+    const route = { params: of({ participationId: '5' }) } as any as ActivatedRoute;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [VcsRepositoryAccessLogViewComponent],
+            providers: [
+                { provide: ActivatedRoute, useValue: route },
+                { provide: ProgrammingExerciseParticipationService, useClass: MockProgrammingExerciseParticipationService },
+                { provide: ProgrammingExerciseService, useClass: MockProgrammingExerciseService },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: AlertService, useClass: MockAlertService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(VcsRepositoryAccessLogViewComponent);
+                component = fixture.componentInstance;
+                programmingExerciseParticipationService = fixture.debugElement.injector.get(ProgrammingExerciseParticipationService);
+
+                repositoryVcsAccessLogSpy = jest.spyOn(programmingExerciseParticipationService, 'getVcsAccessLogForRepository').mockReturnValue(of(mockVcsAccessLog));
+                participationVcsAccessLogSpy = jest.spyOn(programmingExerciseParticipationService, 'getVcsAccessLogForParticipation').mockReturnValue(of(mockVcsAccessLog));
+            });
+    });
+
+    it('should load participation vcs access log', () => {
+        component.ngOnInit();
+        expect(component).toBeTruthy();
+        expect(participationVcsAccessLogSpy).toHaveBeenCalled();
+        expect(component.vcsAccessLogEntries).toHaveLength(2);
+        expect(component.vcsAccessLogEntries[0].userId).toBe(userId);
+    });
+
+    it('should load template repository vcs access log', () => {
+        route.params = of({ exerciseId: '10', repositoryType: 'TEMPLATE' });
+        component.ngOnInit();
+        expect(repositoryVcsAccessLogSpy).toHaveBeenCalled();
```

Comment:
Consider replacing `toHaveBeenCalled` with the more specific `toHaveBeenCalledOnce` for these two occurrences